### PR TITLE
[DMN Designer] New boxed expression editor - Support undo/redo in Boxed Expression Editor

### PR DESCRIPTION
**Pull request by danielzhe**
Vsix v2.0: https://drive.google.com/file/d/1ELkPVzwHfEyDGwcDUqh0jgQuV9vo-fZt/view?usp=sharing

### CODE REVIEW

**Comment by danielzhe** [comment]
Ready, @karreiro, @jomarko , @tiagobento

**Comment by jomarko** [comment]
@danielzhe I found an issue, if user works with multiple nodes expressions in parallel. See more details in https://issues.redhat.com/browse/KOGITO-4195?focusedCommentId=19094068&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-19094068.

**Comment by danielzhe** [comment]
Thank you, @jomarko ! Looks like the issue you reported is present in main. I think this is very old issue, btw.
I also noticed that maybe you are testing the old boxed expression editor, because I realized that they are not available in VSCode yet because of [this PR](https://github.com/kiegroup/kogito-tooling/pull/631) that is not merged. Sorry about that. So I build it locally and I'm providing the Vsix for the reviews to test: https://drive.google.com/file/d/1RM-OjNv6WTDHjt-XMDefmXVXTUaD941Z/view?usp=sharing

File: kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/editors/expressions/ExpressionEditorViewImpl.java [hunk 1]
```diff
@@ -462,46 +504,6 @@ private String getTypeRef() {
...
-        }
-    }
-
-    @SuppressWarnings("rawtypes")
-    private void updateCanvasNodeName() {
-        final AbstractCanvasHandler canvasHandler = (AbstractCanvasHandler) sessionManager.getCurrentSession().getCanvasHandler();
-        final CompositeCommand.Builder<AbstractCanvasHandler, CanvasViolation> commandBuilder = new CompositeCommand.Builder<>();
-        final Element element = canvasHandler.getGraphIndex().get(nodeUUID);
-        if (element.getContent() instanceof Definition) {
-            final Definition definition = (Definition) element.getContent();
```
Comment (vpellegrino) [inline kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/editors/expressions/ExpressionEditorViewImpl.java:495]: Can you please help me to understand how these methods are mapped with the new implementation?

**Review by jomarko — COMMENTED** [review]
@danielzhe thank you for the vsix, 

however I am not able to verify manually scenarios , where user work with more expressions in parallel. See my comment https://github.com/kiegroup/kogito-tooling/pull/631#pullrequestreview-771337733

**Review by jomarko — CHANGES_REQUESTED** [review]
I found out that decision table - add column operation can not be undone.

1. open a DMN file
2. add a decision node
3. open its expression editor
4. activate new expression editor
5. set expression as decision table
6. insert output column (I was able to reproduce with any kind of a column)
7. try to undo as 'Ctrl+Z' - **the added column will be still present**

**Comment by jomarko** [comment]
@danielzhe @vpellegrino reading the comments here, now I got confused, should I use for testing of a #120 PR provided VSIX or a dmn-kogito-testing webapp?

**Comment by vpellegrino** [comment]
> @danielzhe @vpellegrino reading the comments here, now I got confused, should I use for testing of a #120 PR provided VSIX or a dmn-kogito-testing webapp?

I think you should use the VSIX that @danielzhe has provided to you.
About the `dmn-webapp-kogito-testing`, my comment (https://github.com/kiegroup/kogito-editors-java/pull/120#pullrequestreview-771328922) was just to thank @danielzhe for enabling undo/redo feature also on that webapp, since previously it was not present.

**Comment by jomarko** [comment]
@vpellegrino ok, then we are fine. I used VSIX.

File: kie-wb-common-dmn/kie-wb-common-dmn-api/src/main/java/org/kie/workbench/common/dmn/api/definition/model/DecisionRule.java [hunk 2]
```diff
@@ -102,19 +109,21 @@ public boolean equals(final Object o) {
 
         final DecisionRule that = (DecisionRule) o;
 
-        if (id != null ? !id.equals(that.id) : that.id != null) {
-            return false;
+        if (!ignoreId) {
+            if (id != null ? !id.equals(that.id) : that.id != null) {
+                return false;
```
Comment (yesamer) [inline kie-wb-common-dmn/kie-wb-common-dmn-api/src/main/java/org/kie/workbench/common/dmn/api/definition/model/DecisionRule.java:114]: @danielzhe These 2 if can be merged

File: kie-wb-common-dmn/kie-wb-common-dmn-client/src/test/java/org/kie/workbench/common/dmn/client/editors/expressions/util/UserActionCheckerTest.java [hunk 3]
```diff
@@ -0,0 +1,905 @@
...
+    public void setup() {
+        checker = spy(new UserActionChecker());
+    }
+
+    @Test
+    public void testIsUserAction_WhenIsDecisionTable() {
+
+        final DecisionTableProps decisionTableProps = new DecisionTableProps(NAME,
+                                                                             "data type",
+                                                                             "hit policy",
```
Comment (jomarko) [inline kie-wb-common-dmn/kie-wb-common-dmn-client/src/test/java/org/kie/workbench/common/dmn/client/editors/expressions/util/UserActionCheckerTest.java:64]: it seems we forget to use DATA_TYPE and HIT_POLICY on some places in this test class?

**Review by jomarko — COMMENTED** [review]
One comment to code, I would like to recheck manually after #155 merge.

**Pull request closed without merging**